lib/puppet/application/resource.rb

```diff
@@ -109,6 +110,10 @@ def help
         Output found resources in yaml format, suitable to use with Hiera and
         create_resources.
 
+      * --fail-when-not-found:
+        Fails and returns an exit code of 1 if the resource is not found, valid only
+        for Windows and SystemD services.
+
```

Comment:
I was under the impression this applies to any resource, not just services, for example, this now returns -1:

```
❯ bundle exec puppet resource file /tmp ensure=file --fail-when-not-found
Notice: /File[/tmp]: Not removing directory; use 'force' to override
Notice: /File[/tmp]: Not removing directory; use 'force' to override
Error: Could not set 'file' on ensure: Is a directory @ rb_sysopen - /tmp
Error: Could not set 'file' on ensure: Is a directory @ rb_sysopen - /tmp
Wrapped exception:
Is a directory @ rb_sysopen - /tmp
Error: /File[/tmp]/ensure: change from 'directory' to 'file' failed: Could not set 'file' on ensure: Is a directory @ rb_sysopen - /tmp
Error: Could not run: Failed to manage resource File[/tmp]

❯ echo $?
1
```

Though the Windows and SystemD service providers can now report on a non-existent versus stopped service

Reply:
Oh good call that for sure changed over the course of my implementation of this.